src/types/iterator.rs

```diff
@@ -33,11 +34,7 @@ impl PyIterator {
     ///
     /// Equivalent to Python's built-in `iter` function.
     pub fn from_object(obj: &PyAny) -> PyResult<&PyIterator> {
-        Self::from_object2(Py2::borrowed_from_gil_ref(&obj)).map(|py2| {
-            // Can't use into_gil_ref here because T: PyTypeInfo bound is not satisfied
-            // Safety: into_ptr produces a valid pointer to PyIterator object
-            unsafe { obj.py().from_owned_ptr(py2.into_ptr()) }
-        })
+        Self::from_object2(Py2::borrowed_from_gil_ref(&obj)).map(Py2::into_gil_ref)
```

Comment:
This is an example of where this adjustment simplifies interchange between the gil-ref and `Py2` APIs; there are more cases in #3572 which can be simplified also.